compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -1494,6 +1494,18 @@ public BType createInvokableType(List<? extends BLangVariable> paramVars,
 
         boolean foundDefaultableParam = false;
         List<String> paramNames = new ArrayList<>();
+        if (Symbols.isFlagOn(flags, Flags.ANY_FUNCTION)) {
+            BInvokableType bInvokableType = new BInvokableType(paramTypes, null, symTable.anyOrErrorType, null);
+            bInvokableType.flags = flags;
+            BInvokableTypeSymbol tsymbol = Symbols.createInvokableTypeSymbol(SymTag.FUNCTION_TYPE, flags,
+                                                                             env.enclPkg.symbol.pkgID, bInvokableType,
+                                                                             env.scope.owner, location, SOURCE);
+            tsymbol.params = params;
```

Comment:
Can we pass null instead of `paramTypes` at L1498 and set null here too? IMO, it'll then clearly differentiate the zero params scenario and the absence of params scenario.

Reply:
Fixed.